packages/realm/src/Realm.ts

```diff
@@ -599,7 +599,7 @@ export class Realm {
   public readonly syncSession: SyncSession | null;
 
   private schemaExtras: RealmSchemaExtra = {};
-  private classes: ClassMap;
+  private classes!: ClassMap;
```

Comment:
I'm almost thinking that this drawback from deferring to a method at line 688 isn't worth it. As we don't expect to have to initialize the class map more than twice and it's not too complex, I would probably suggest duplicating and inlining it instead.

Reply:
You make a good point, I'll inline those calls 👍

P.S. would be nice to add something in function signatures that tells TS that a certain property will be set. Similar to `asserts` or `is`. That would remove the risks of having to use `!` as in the above case.